### DISCUSSION TITLE
search is present on live server but disappeared in offline build

### DISCUSSION THREAD

**Discussion by jablka** [squidfunk/mkdocs-material#6605] [Q&A] [answered]
Hi, 
I'm new here, trying to build a first sample site (following your documentation), but encountering a problem:

The _search input_ is present on live server, but disappears when I run offline build.
I also added to the `mkdocs.yml` following:
```
site_name: My Docs
theme:
  name: material
plugins:
  - search
```
but it doesn't solve the issue...

I see the search is missing also on James Willett tutorial sample:
https://james-willett.github.io/mkdocs-material-youtube-tutorial/
🤔

How to solve it?

**Comment by kamilkrzyskow** [comment]
Hello @jablka,
for offline builds please use the `offline` plugin: https://squidfunk.github.io/mkdocs-material/plugins/offline/
and also please keep in mind the limitations: https://squidfunk.github.io/mkdocs-material/plugins/offline/#limitations
```yaml
site_name: My Docs
theme:
  name: material
plugins:
  - search
  - offline
```

As for https://james-willett.github.io/mkdocs-material-youtube-tutorial/ not having search is because it doesn't include `search` in the plugins:
https://github.com/james-willett/mkdocs-material-youtube-tutorial/blob/a7ed9003c31776fe36157ca667ea3a1958e97740/mkdocs.yml#L30-L31

Further reading: https://www.mkdocs.org/user-guide/configuration/#plugins

**Reply by jablka** [reply]
thank you, it works!

**Comment by docopshabr** [comment]
Hi!

I made a project like this, but unfortunately the search field disappears if the internet goes down. Am I right that the search should work even without internet access?

**Reply by squidfunk** [reply]
Yes, as long as you're using our integrated search plugin does not need internet access when you run it locally. If you load a site from a deployed installation (e.g. GitHub Pages) and in the middle of loading, internet goes down, the site might be rendered, but loading of the search index was interrupted. In that case, search will not work.

**Reply by docopshabr** [reply]
The situation is as follows))):

1. I created a new project with this configuration.

```
site_name: My Docs
theme:
  name: material
plugins:
  - search
  - offline
```

2. I generated a site with `mkdocs build` command.
3. I opened the `index.html` from the `site` folder and the search works well.
4. I turned off the internet and the search field is gone.

Do I understand correctly that with offline plugin we can continue to use search even if our machine is not connected to internet?

**Reply by squidfunk** [reply, marked as answer]
Ah I see. Yes, the problem is that the font cannot be loaded (since it's referenced from Google), and the `iframe-worker` shim from unpkg.com, which you'll see if you open the browser console. To solve it, just add the privacy plugin, which will allow full offline capability. This is also [explained here](https://squidfunk.github.io/mkdocs-material/plugins/offline/#when-to-use-it):

``` yaml
site_name: My Docs
theme:
  name: material
plugins:
  - search
  - offline
  - privacy
```

The logic of downloading and inlining external assets, and preparing the site for offline use is split among the offline and privacy plugin, since we would otherwise end up with duplicated functionality. We'll evaluate in the future how we can make this easier to configure.

**Reply by docopshabr** [reply]
Thanks a lot for your support!!! It's works well!

**Reply by jeremy-feng** [reply]
Thanks for your discussion. I've encountered the exact same issue. Unfortunately, I don't have internet access on my machine, which causes the process to halt at the following step:

```bash
INFO    -  Downloading external file: https://unpkg.com/iframe-worker/shim
```

Is it possible to download the file on a different machine and then transfer it to the offline machine? If so, could you please advise where I should place the file? Thanks!

**Reply by squidfunk** [reply]
Yes, you can build on one machine with internet access and then just copy the `.cache` folder to the other machine.

**Reply by jeremy-feng** [reply]
Thanks a lot.

**Reply by squidfunk** [reply]
Another idea: we generally recommend to add `.cache` to `.gitignore`, but you can also refine that to keep specific parts of the cache in the repository, e.g. for the privacy plugin add everything to `.gitignore` except `.cache/plugins/privacy`.
